Tests/BSONTests/ExtendedJSONConversionTests.swift

```diff
@@ -103,6 +103,41 @@ open class ExtendedJSONConversionTestCase: BSONTestCase {
         expect(String(data: fooEncoded, encoding: .utf8)).to(contain("false"))
     }
 
+    func testOutputFormatting() throws {
+        let encoder = ExtendedJSONEncoder()
+        let input: BSONDocument = ["topLevel": ["hello": "world"]]
+
+        let defaultFormat = String(data: try encoder.encode(input), encoding: .utf8) ?? ""
```

Comment:
nit: I don't think any of the `?? ""` are necessary, Nimble will let us pass in optionals and compare them to the expected non-optional strings

Reply:
fixed